language/move-prover/boogie-backend/src/bytecode_translator.rs

```diff
@@ -1095,13 +1245,182 @@ impl<'env> FunctionTranslator<'env> {
+                                                baseline_flag,
+                                            )
+                                            .unwrap()
+                                            == Bitwise
+                                    } else {
+                                        false
+                                    };
+                                    emitln!(
+                                        writer,
+                                        "call {}({});",
+                                        boogie_function_bv_name(&callee_env, inst, bv_flag),
+                                        args_str
+                                    );
+                                } else {
+                                    emitln!(
+                                        writer,
+                                        "call {}({});",
+                                        boogie_function_name(&callee_env, inst),
+                                        args_str
+                                    );
+                                }
+                            } else if module_env
+                                .get_name()
+                                .name()
+                                .display(env.symbol_pool())
+                                .to_string()
+                                == "vector"
+                            {
+                                let callee_name = callee_env
+                                    .get_name()
+                                    .display(self.fun_target.global_env().symbol_pool())
+                                    .to_string();
+                                let dest_bv_flag = if !dests.is_empty() && targeted && fun_verified
+                                {
+                                    *global_state
+                                        .get_temp_index_oper(
+                                            caller_mid,
+                                            caller_fid,
+                                            dests[0],
+                                            baseline_flag,
+                                        )
+                                        .unwrap()
+                                        == Bitwise
+                                } else {
+                                    false
+                                };
+                                let mut bv_flag = if !srcs.is_empty() && targeted && fun_verified {
+                                    *global_state
+                                        .get_temp_index_oper(
+                                            caller_mid,
+                                            caller_fid,
+                                            srcs[0],
+                                            baseline_flag,
+                                        )
+                                        .unwrap()
+                                        == Bitwise
```

Comment:
Can't we avoid doing the same thing here again? And can't we factor this out into an independent function? This code in fact looks rather hacky right now.